app/controllers/tournaments_controller.rb

```
@@ -113,6 +123,41 @@ def create
     end
   end
 
+  def create_demo
+    authorize Tournament
+
+    # We will do basic validation here, while farming out a successful creation to the library code.
+    errors = {}
+
+    errors[:name] = 'You must provide a name for the tournament' if params[:tournament][:name].blank?
+
+    if params[:tournament][:num_players].blank? || !params[:tournament][:num_players].match?(/^\d+$/)
+      errors[:num_players] = 'You must provide a number of players'
+    end
+    if params[:tournament][:num_first_round_byes].present? &&
+       !params[:tournament][:num_first_round_byes].match?(/^\d+$/)
+      errors[:num_first_round_byes] = 'Number of byes must be a number'
+    end
+
+    if errors.empty?
```

I personally prefer early returns / fail early. 

I don't know if there are any Ruby best practices for that, but I would suggest negating the condition, which means the error case is first with early return and the happy path is last.

Not that important because it's only one if-else, just personal preference.